Compile warning in DarwinProcess.c when debug is enabled

```
darwin/DarwinProcess.c:499:39: warning: comparison of integers of different signs: 'pid_t' (aka 'int') and 'uint64_t' (aka 'unsigned long long') [-Wsign-compare]
  499 |       assert(Process_getPid(tprocess) == tid);
      |              ~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/assert.h:75:25: note: expanded from macro 'assert'
   75 |     (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __ASSERT_FILE_NAME, __LINE__, #e) : (void)0)
      |                         ^
```

OS: macOS Sequoia 15.3
Compiler:

```
Apple clang version 16.0.0 (clang-1600.0.26.6)
Target: arm64-apple-darwin24.3.0
```

Reproducible also in CI: https://github.com/Explorer09/htop-1/actions/runs/13082758381/job/36509406353

--- a/darwin/DarwinProcess.c
+++ b/darwin/DarwinProcess.c
@@ -496,7 +496,11 @@ void DarwinProcess_scanThreads(DarwinProcess* dp, DarwinProcessTable* dpt) {
          continue;
       }
 
-      assert(Process_getPid(tprocess) == tid);
+      pid_t tprocessPid = Process_getPid(tprocess);
+      assert(tprocessPid >= 0);
+      assert((uint64_t)tprocessPid == tid);
+      (void)tprocessPid;
+
       Process_setParent(tprocess, pid);
       Process_setThreadGroup(tprocess, pid);
       tprocess->super.show       = true;

